lib/ansible/galaxy/collection.py

```diff
@@ -1070,46 +1167,128 @@ def _build_dependency_map(collections, existing_collections, b_temp_path, apis,
     return dependency_map
 
 
+def _collections_from_scm(collection, requirement, b_temp_path, force, parent=None):
+    """
+    Returns a list of collections found in the repo. If there is a galaxy.yml in the collection then just return
+    the specific collection. Otherwise, check each top-level directory for a galaxy.yml.
+
+    :param collection: URI to a git repo
+    :param requirement: The version of the artifact
+    :param b_temp_path: The temporary path to the archive of a collection
+    :param force: Whether to overwrite an existing collection or fail
+    :param parent: The name of the parent collection
+    :return: List of CollectionRequirement objects
+    """
+
+    reqs = []
+    name, version, path, fragment = parse_scm(collection, requirement)
+    b_repo_root = to_bytes(name, errors='surrogate_or_strict')
+
+    if fragment:
+        b_fragment = to_bytes(fragment, errors='surrogate_or_strict')
+        b_collection_path = os.path.join(b_temp_path, b_repo_root, b_fragment)
+    else:
+        b_collection_path = os.path.join(b_temp_path, b_repo_root)
+
+    b_galaxy_path = os.path.join(b_collection_path, b'galaxy.yml')
+
+    err = "{0} appears to be an SCM collection source, but the required galaxy.yml was not found. " \
+          "Append #path/to/collection/ to your URI (before the comma separated version, if one is specified) " \
+          "to point to a directory containing the galaxy.yml or directories of collections".format(collection)
```

Comment:
Usually, I discourage using EOL escaping because things like
```python
st = 'some words' \

'separate'
```
are valid. And `'separate'` in this example wouldn't be a part of `st`.

Reply:
good point